src/ng-select/ng-dropdown-panel.component.ts

```diff
@@ -0,0 +1,265 @@
+import {
+    Component,
+    OnDestroy,
+    Renderer2,
+    ElementRef,
+    Input,
+    EventEmitter,
+    Output,
+    ViewChild,
+    SimpleChanges,
+    NgZone,
+    TemplateRef,
+    forwardRef,
+    Inject,
+    ViewEncapsulation,
+    ChangeDetectionStrategy,
+} from '@angular/core';
+
+import { VirtualScrollService } from './virtual-scroll.service';
+import { NgOption } from './ng-select.types';
+import { NgSelectComponent, DropdownPosition } from './ng-select.component';
+import { ItemsList } from './items-list';
+import { WindowService } from './window.service';
+
+@Component({
+    providers: [VirtualScrollService],
```

Comment:
Here each component instance will have separate instance of `VirtualScrollService`, not sure if this is a good idea